core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -88,3 +121,55 @@ fn pre_boojum_into_token(l1_batch_commit_with_metadata: &L1BatchWithMetadata) ->
         ),
     ])
 }
+
+fn encode_l1_commit(l1_batch_with_metadata: &L1BatchWithMetadata, pubdata: Token) -> Vec<Token> {
```

Comment:
Looks like you're only using `header` and `metadata` fields in this function. So, you can slightly rework this function:

```rust
fn encode_l1_commit(header: &L1BatchHeader, metadata: &L1BatchMetadata, pubdata_input: Option<&L1BatchWithMetadata>) -> Vec<Token>
```

This way you won't need separate `validium_mode_l1_commit_data` and `rollup_mode_l1_commit_data`